agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -285,6 +285,14 @@ private static void overlayRpConfiguration(Configuration config, RpConfiguration
         if (rpConfiguration.ignoreRemoteParentNotSampled != null) {
             config.preview.ignoreRemoteParentNotSampled = rpConfiguration.ignoreRemoteParentNotSampled;
         }
+        if (isTrimEmpty(config.role.name)) {
+            // only use rp configuration role name as a fallback, similar to WEBSITE_SITE_NAME
+            config.role.name = rpConfiguration.role.name;
+        }
+        if (isTrimEmpty(config.role.instance)) {
+            // only use rp configuration role name as a fallback, similar to WEBSITE_INSTANCE_ID
+            config.role.instance = rpConfiguration.role.instance;
```

Comment:
Azure Spring Cloud will set the role name from applicationinsights-rp.json by default. But we should allow customer to override the default behavior by env variables like ROLE_NAME and ROLE_INSTANCE.

Currently looks rpConfiguration will always win, right ?